python/composio/tools/env/docker/shell.py

```diff
@@ -200,16 +205,23 @@ def setup(self) -> None:
             else:
                 raise ValueError(f"Invalid command type: {cmd_file.cmd_type}")
 
-    def exec(self, cmd: str) -> t.Dict:
-        """Execute command on container."""
+    def _write(self, cmd: str) -> None:
+        """Write command to console."""
         try:
             stdin = t.cast(t.IO[str], self._process.stdin)
             os.write(stdin.fileno(), self.sanitize_command(cmd=cmd))
             stdin.flush()
         except BrokenPipeError as e:
             # TODO: Handle this as framework error
             raise RuntimeError(str(e)) from e
-        return self._read()
+
+    def exec(self, cmd: str) -> t.Dict:
+        """Execute command on container."""
+        self._write(cmd=cmd)
+        return {
+            **self._read(),
+            "exit_code": self._get_exit_code(),
```

Comment:
exit-code should be a constant defined above, as we are using this key at other places as well